packages/block/src/block.ts

```diff
@@ -11,8 +11,8 @@ import {
   Transaction,
   Capability,
 } from '@ethereumjs/tx'
-import { BlockHeader } from './header'
-import { BlockData, BlockOptions, JsonBlock, BlockBuffer, Blockchain } from './types'
+import { BlockHeader } from './header.js'
+import { BlockData, BlockOptions, JsonBlock, BlockBuffer, Blockchain } from './types.js'
```

Comment:
Rather than doing this, you could use a compiler plugin to automatically do it.  There is much disagreement on what is "correct" here, but I'm of the opinion that you should leave `./header` and let the compiler add the extension.  The reason for this is because you aren't actually importing `./header.js`, you are importing `./header.ts` and then after transpilation it should turn into `./header.js`.

If you want to have the compiler automatically add the extension, rather than adding them manually, you can use the extension here:
https://github.com/Zoltu/typescript-transformer-append-js-extension/